web/src/test/java/org/cbioportal/web/DataAccessTokenControllerTest.java

```diff
@@ -30,69 +30,73 @@
 import org.cbioportal.service.DataAccessTokenService;
-import org.cbioportal.weblegacy.DataAccessTokenControllerConfig;
-import org.cbioportal.weblegacy.exception.DataAccessTokenNoUserIdentityException;
-import org.cbioportal.weblegacy.exception.DataAccessTokenProhibitedUserException;
+import org.cbioportal.service.exception.DataAccessTokenNoUserIdentityException;
+import org.cbioportal.service.exception.DataAccessTokenProhibitedUserException;
 import org.junit.Assert;
 import org.junit.Before;
 import org.junit.Test;
 import org.junit.runner.RunWith;
 import org.mockito.Matchers;
 import org.mockito.Mockito;
 import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
 import org.springframework.http.HttpStatus;
-import org.springframework.http.MediaType;
 import org.springframework.http.ResponseEntity;
 import org.springframework.security.authentication.UsernamePasswordAuthenticationToken;
 import org.springframework.security.core.Authentication;
 import org.springframework.security.core.GrantedAuthority;
+import org.springframework.security.core.userdetails.User;
 import org.springframework.test.context.ContextConfiguration;
 import org.springframework.test.context.junit4.SpringJUnit4ClassRunner;
 import org.springframework.test.context.web.WebAppConfiguration;
 
 @RunWith(SpringJUnit4ClassRunner.class)
-//@WebAppConfiguration
-@ContextConfiguration(classes = {DataAccessTokenControllerConfig.class})
+@WebAppConfiguration
+@ContextConfiguration("/applicationContext-web.xml")
+@Configuration
 public class DataAccessTokenControllerTest {
+     @Autowired
+     DataAccessTokenService tokenService;
 
-    @Autowired
-    private DataAccessTokenService tokenService;
+     @Autowired
+     DataAccessTokenController dataAccessTokenController;
 
-    @Autowired
-    private DataAccessTokenController dataAccessTokenController;
+     @Bean
+     public DataAccessTokenService tokenService() {
+         return Mockito.mock(DataAccessTokenService.class);
+     }
+
+     @Bean
+     public DataAccessTokenController dataAccessTokenController() {
+         return new DataAccessTokenController();
+     }
 
     public static final String API_TEST_SUBJECT = "testSubject";
     public static final String NON_API_TEST_SUBJECT = "anonymousUser";
     public static final String MOCK_TOKEN_STRING = "MockedTokenString";
     public static final DataAccessToken MOCK_TOKEN_INFO = new DataAccessToken(MOCK_TOKEN_STRING);
-    
+
     @Before
     public void setup() {
         Mockito.reset(tokenService);
     }
-   
+
     protected Authentication makePrincipal(String username, boolean isAuthentic) {
-        PortalUserDetails portalUserDetails = makePortalUserDetails(username);
+        User user = new User(username, "unused", new ArrayList<GrantedAuthority>());
```

Comment:
this avoids bringing in dependencies from the security module (because we don't need PortalUserDetails)

Reply:
correct